src/java/picard/illumina/ClusterDataToSamConverter.java

```diff
@@ -134,12 +169,29 @@ private SAMRecord createSamRecord(final ReadData readData, final String readName
             sam.setAttribute(SAMTag.BC.name(), unmatchedBarcode);
         }
 
-        if (molecularIndex != null) {
+        if (!molecularIndexes.isEmpty()) {
             //TODO: add RX and QX to the list of SAMTags and change this. initial discussion
             //TODO: here: https://github.com/broadinstitute/picard/issues/287
             //TODO: HTS-spec issue: https://github.com/samtools/hts-specs/issues/109
-            sam.setAttribute("RX", molecularIndex);
-            sam.setAttribute("QX", molecularIndexQ);
+            sam.setAttribute(this.molecularIndexTag, String.join("", molecularIndexes));
+            sam.setAttribute(this.molecularIndexQualityTag, String.join("", molecularIndexQualities));
+            if (!this.tagPerMolecularIndex.isEmpty() && molecularIndexes.size() > 1) {
+                if (tagPerMolecularIndex.size() < molecularIndexes.size()) {
+                    // fewer tags than indexes, so store the remaining in the last tag
```

Comment:
This is explained both in the definition of the `withTagPerMolecularIndex` method and in the `IlluminaBasecallsToSam`'s `TAG_PER_MOLECULAR_INDEX` option. 

What I *really* want is a read structure that can describe which molecular indexes should go into which tags (with concatenation), and possible ordering in concatenation.  For example `8M{RX}50T50T4M{ZA}4M{RX}` but that is for a different day. 

 Nonetheless, the convention of concatenating the molecular indexes when fewer tags are available is up for debate here.  I am in favor of this behavior.  Do others have thoughts?  @ktibbett @tfenne?

I will at minimum put a warning in place.